x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/network/top_n_flow/helpers.ts

```diff
@@ -68,24 +66,22 @@ const formatTopNFlowEdges = (
     },
   }));
 
-const getFlowTargetFromString = (flowAsString: string) =>
-  flowAsString === 'source' ? FlowTargetSourceDest.source : FlowTargetSourceDest.destination;
-
-const getGeoItem = (result: NetworkTopNFlowBuckets): GeoItem | null =>
+const getGeoItem = (
+  result: NetworkTopNFlowBuckets,
+  flowTarget: FlowTargetSourceDest
+): GeoItem | null =>
   result.location.top_geo.hits.hits.length > 0 && result.location.top_geo.hits.hits[0].fields
     ? {
         geo: formatResponseObjectValues(
           getOr(
             '',
-            `${Object.keys(result.location.top_geo.hits.hits[0].fields)[0].split('.geo')[0]}.geo`,
```

Comment:
This was very silly. If for some reason a non-geo field is in there (ahem, `@timestamp`), the code breaks. We are able to simply pass `flowTarget` from where the function is called.

Reply:
Now, it looks ten times better  🤩